Source/levels/themes.cpp

```diff
@@ -101,28 +101,26 @@ bool CheckThemeObj5(Point origin, int8_t regionId)
 
 bool TFit_Obj5(int t)
 {
-	int skipCandidates = GenerateRnd(5);
-	if (skipCandidates < 0) {
+	const int targetCandidates = GenerateRnd(5);
+	if (targetCandidates < 0) {
 		// vanilla rng can return -3 for GenerateRnd(5), default behaviour is to set the output to 0,0 and return true in this case...
 		themex = 0;
 		themey = 0;
 		return true;
 	}
 
-	for (int yp = 0; yp < MAXDUNY; yp++) {
-		for (int xp = 0; xp < MAXDUNX; xp++) {
-			if (dTransVal[xp][yp] == themes[t].ttval && IsTileNotSolid({ xp, yp }) && CheckThemeObj5({ xp, yp }, themes[t].ttval)) {
-				if (skipCandidates > 0) {
-					skipCandidates--;
-					continue;
-				}
-				themex = xp;
-				themey = yp;
-				return true;
-			}
+	int candidatesFound = 0;
+	for (Point tile : PointsInRectangleRange { { { 0, 0 }, { MAXDUNX, MAXDUNY } } }) {
+		if (dTransVal[tile.x][tile.y] == themes[t].ttval && IsTileNotSolid(tile) && CheckThemeObj5(tile, themes[t].ttval)) {
+			// Use themex/y to keep track of the last candidate area found, in case we end up with fewer candidates than the target
+			themex = tile.x;
+			themey = tile.y;
+			candidatesFound++;
+			if (candidatesFound > targetCandidates)
+				break;

```

Comment:
nit style question: should we return here instead of break?
For me this would be a little bit more clear ("We found our candidate, get the hell out of here" 😓).
Note: you can discard the comment. It's only a style question not functional.

Reply:
Yeah, I was torn. I went with a break because the return value is a different condition, even though it's always true if this gets hit...

Reply:
Needed to rebase anyway so went with the early return this time :D